grapher/core/Grapher.tsx

```diff
@@ -2205,26 +2211,26 @@ export class Grapher
     }
 
     @computed get timeParam() {
-        const formatAsDay = this.table.hasDayColumn
-        if (
-            this.isOnMapTab &&
-            this.map.time !== undefined &&
-            this.hasUserChangedMapTimeHandle
-        )
-            return timeBoundToTimeBoundString(this.map.time, formatAsDay)
+        const { timeColumn } = this.table
+        const formatTime = (t: Time) =>
+            timeBoundToTimeBoundString(
+                t,
+                timeColumn instanceof ColumnTypeMap.Day
+            )
+
+        if (this.isOnMapTab) {
+            return this.map.time !== undefined &&
+                this.hasUserChangedMapTimeHandle
+                ? formatTime(this.map.time)
+                : undefined
+        }
+
         if (!this.hasUserChangedTimeHandles) return undefined
 
-        const [startHandleTime, rightHandleTime] = this.timelineHandleTimeBounds
-        const startTimeBoundString = timeBoundToTimeBoundString(
-            startHandleTime,
-            formatAsDay
+        const [startTime, endTime] = this.timelineHandleTimeBounds.map(
+            formatTime
         )
-        return startHandleTime === rightHandleTime
-            ? startTimeBoundString
-            : `${startTimeBoundString}..${timeBoundToTimeBoundString(
-                  rightHandleTime,
-                  formatAsDay
-              )}`
+        return startTime === endTime ? startTime : `${startTime}..${endTime}`
```

Comment:
@MarcelGerber Because these are TimeBounds so that `Infinity` has a special meaning (`"latest"`), I thought it might be better to keep the logic in here instead of on the Column classes (with the `formatForQueryString` method I proposed).

Reply:
I will merge to fix the bug, but let me know if it seems more reasonable to you to format `TimeBound` on the column class!